src/NTestController/IPlugin.cs

```diff
@@ -22,6 +23,30 @@ public interface IPlugin
         bool Execute();
     }
 
+    public interface IReaderPlugin : IPlugin
+    {
+        string TestInputFile { get; set; }
+        List<Test> Tests { get; }
+    }
+
+    public interface IReporterPlugin : IPlugin
+    {
+        TestQueue TestQueue { get; set; }
+    }
+
+    public interface IExecutorPlugin : IPlugin
+    {
+        IReadOnlyOptions Options { get; set; }

```

Comment:
Maybe all the plugins should have an Options property (not sure if they need it for the Verbose level or not)?  In which case, putting it in IPlugin might be better.